updater/lib/dependabot/update_files_command.rb

```diff
@@ -10,20 +10,19 @@ class UpdateFilesCommand < BaseCommand
     def perform_job
       # We expect the FileFetcherCommand to have been executed beforehand to place
       # encoded files and commit information in the environment, so let's retrieve
-      # and decode them into an object.
-
-      # TODO: Parse the dependency files when instantiated
-      #
-      # We can pull the error handling for parser exceptions up into this class to
-      # completely remove the concern from Dependabot::Updater.
-      #
-      # This should happen separately to introducing the class as a shim.
-      #
-      # See: updater/lib/dependabot/dependency_snapshot.rb:52
-      dependency_snapshot = Dependabot::DependencySnapshot.create_from_job_definition(
-        job: job,
-        job_definition: Environment.job_definition
-      )
+      # them, decode and parse them into an object that knows the current state
+      # of the project's dependencies.
+      begin
+        dependency_snapshot = Dependabot::DependencySnapshot.create_from_job_definition(
+          job: job,
+          job_definition: Environment.job_definition
+        )
+      rescue StandardError => e
+        handle_parser_error(e)
```

Comment:
I'm split on this as I originally created that class to contain these two long methods, but because it is namespaced within the Updater it felt right to remove the concern from it as being out of bounds.

I'm still a bit uncertain of the long term place for that helper, ideally we'd push the need to do the dance to get from `DependabotError` class to `error-details` hash to be a convention where every error implemented the "presentation" hash directly but that's a big lift and pretty outside the critical path.